src/power_grid_model/utils.py

```diff
@@ -398,3 +404,61 @@ def self_test():
             print("Self test finished.")
         except Exception as e:
             raise PowerGridError from e
+
+
+def _make_test_case(
+    save_path: Path,
+    input_data: SingleDataset,
+    params: dict,
+    output_data: Dataset,
+    output_dataset_type: DatasetType,
+    update_data: Dataset | None = None,
```

Comment:
probably good to make everything kwargs for future compatibility
```suggestion
def _make_test_case(
    *,
    save_path: Path,
    input_data: SingleDataset,
    params: dict,
    output_data: Dataset,
    output_dataset_type: DatasetType,
    update_data: Dataset | None = None,
```